src/lib/dusk.rs

```diff
@@ -0,0 +1,217 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use core::cmp::Ordering;
+use std::fmt;
+use std::num::ParseFloatError;
+use std::ops::{Add, Deref, Div, Mul, Sub};
+use std::str::FromStr;
+
+use rusk_abi::dusk;
+
+/// The underlying unit of Dusk
+pub type Lux = u64;
+
+pub(crate) const MIN: Dusk = Dusk(dusk::LUX);
+pub(crate) const MAX: Dusk = Dusk(dusk::dusk(f64::MAX));
+
+/// Denomination for DUSK
+#[derive(Copy, Clone, Eq)]
+pub struct Dusk(Lux);
+
+/// Core ops
+/// Implementations of Addition, Subtraction, Multiplication,
+/// Division, and Comparison operators for Dusk
+
+/// Addition
+impl Add for Dusk {
+    type Output = Self;
+    fn add(self, other: Self) -> Self {
+        Self(self.0 + other.0)
+    }
+}
+
+/// Subtraction
+impl Sub for Dusk {
+    type Output = Self;
+    fn sub(self, other: Self) -> Self {
+        Self(self.0 - other.0)
+    }
+}
+
+/// Multiplication
+impl Mul for Dusk {
+    type Output = Self;
+    fn mul(self, other: Self) -> Self {
+        let a = dusk::from_dusk(self.0);
+        let b = dusk::from_dusk(other.0);
+        Self(dusk::dusk(a * b))
+    }
+}
+
+/// Division
+impl Div for Dusk {
+    type Output = Self;
+    fn div(self, other: Self) -> Self {
+        Self(dusk::dusk(self.0 as f64 / other.0 as f64))
+    }
+}
+
+/// Equality
+impl PartialEq for Dusk {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+impl PartialEq<Lux> for Dusk {
+    fn eq(&self, other: &Lux) -> bool {
+        self.0 == *other
+    }
+}
+impl PartialEq<f64> for Dusk {
+    fn eq(&self, other: &f64) -> bool {
+        self.0 == dusk::dusk(*other)
+    }
+}
+
+/// Comparison
+impl Ord for Dusk {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.0.cmp(&*other)
+    }
+}
+
+impl PartialOrd for Dusk {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.0.partial_cmp(&other.0)
+    }
+}
+impl PartialOrd<Lux> for Dusk {
+    fn partial_cmp(&self, other: &Lux) -> Option<Ordering> {
+        self.0.partial_cmp(other)
+    }
+}
+impl PartialOrd<f64> for Dusk {
+    fn partial_cmp(&self, other: &f64) -> Option<Ordering> {
+        self.0.partial_cmp(&dusk::dusk(*other))
+    }
+}
+
+/// Conversion ops
+/// Conveninent conversion of primitives to and from Dusk
+
+/// Floats are used directly as Dusk value
+impl From<f64> for Dusk {
+    fn from(val: f64) -> Self {
+        Self(dusk::dusk(val))
+    }
+}
```

Comment:
If we can convert *any* `f64` to `Dusk`, then `MIN` doesn't make sense. Since `MIN` is one Lux, that would means I shouldn't be able to convert `0.0` or something that will end up to an amount that is less than one Lux.

I think we should either remove the capability to convert from `f64` (but still be able to convert any `Dusk` to `f64`) or remove the `MIN` from the type. But it depends from what is used for. 

You also might want to implement the conversion for `&Dusk` type.